packages/next/client/route-announcer.tsx

```diff
@@ -36,14 +36,9 @@ export const RouteAnnouncer = () => {
       if (previouslyLoadedPath.current === asPath) return
       previouslyLoadedPath.current = asPath
 
-      if (document.title) {
-        setRouteAnnouncement(document.title)
-      } else {
-        const pageHeader = document.querySelector('h1')
-        const content = pageHeader?.innerText ?? pageHeader?.textContent
+      const pageHeader = document.querySelector('h1')
 
-        setRouteAnnouncement(content || asPath)
-      }
+      setRouteAnnouncement(document?.title || (pageHeader?.innerText ?? pageHeader?.textContent) || asPath)
```

Comment:
You should always write the code **without** tricks, just let modification does its job (and both swc/terser are pretty good at it).